src/Network/ABCI.hs

```diff
@@ -20,47 +21,37 @@ import           Network.ABCI.Internal.Wire           as Wire
 import           Network.ABCI.Types.App               (App (..))
 import qualified Network.ABCI.Types.Messages.Request  as Request
 import qualified Network.ABCI.Types.Messages.Response as Response
-import           Network.Socket                       (SockAddr)
 import qualified Proto.Types                          as PT
-import           UnliftIO                             (MonadUnliftIO)
 
 
+-- | Default ABCI app network settings for serving on localhost at the
+-- standard port.
+defaultLocalSettings :: ServerSettings
+defaultLocalSettings = serverSettings 26658 $ fromString "127.0.0.1"
 
--- | Default ABCI app network settings.
-defaultSettings :: ServerSettings
-defaultSettings = serverSettings 26658 $ fromString "127.0.0.1"
-
--- | Serve an ABCI application with custom 'ServerSettings'
+-- | Serve an ABCI application with custom 'ServerSettings' and a custom
+-- action to perform on aquiring the socket resource.
 serveAppWith
-  :: ( MonadIO m
-     , MonadBaseControl IO m
-     , MonadUnliftIO m
-     )
-  => ServerSettings
-  -> (SockAddr -> m (App m))
-  -> m ()
-serveAppWith cfg mkApp = runGeneralTCPServer cfg $ \appData -> do
-  app <- mkApp (appSockAddr appData)
-  runConduit (setupConduit app appData)
+  :: ServerSettings
+  -> App IO
+  -> (AppData -> IO ())
```

Comment:
I suspect when AppData is useful, you would need it in the App itself, and with this signature, it would be hard to obtain it in the App (you would need to use some MVar to put AppData into, and then read from it in the App). 
Is there any disadvantage for `AppData -> IO (App IO)`?

Reply:
It might be too early to tell, but I somehow doubt that this `AppData` would be useful at the level of `App` creation. You can find its [definition](https://hackage.haskell.org/package/streaming-commons-0.2.1.0/docs/src/Data.Streaming.Network.Internal.html#AppData) in the [streaming-commons](https://hackage.haskell.org/package/streaming-commons-0.2.1.0) lib. If the `App` is using this, it means it's reading / writing from the socket directly without the preprocessing conduits, or possibly close the connection itself. I'm not sure we need any of these features, but who knows. Delay decision till later? 

The reason I even wanted to expose it at this level is for testing, since I think we can write messages to the socket directly with [this](https://hackage.haskell.org/package/streaming-commons-0.2.1.0/docs/Data-Streaming-Network.html#v:writeLens)

Reply:
good point take a look at  #26